npm-scripts.js

```diff
@@ -365,3 +400,62 @@ function installMsysMake()
 
 	executeCmd(`${String(res.stdout).trim()} worker\\scripts\\getmake.py`);
 }
+
+function ensureDir(dir)
+{
+	if (!fs.existsSync(dir))
+	{
+		fs.mkdirSync(dir, { recursive: true });
+	}
+}
+
+function createTar(files, dest)
+{
+	const tar = require('tar');
+
+	tar.create(
+		{
+			gzip : true
+		},
+		files
+	).pipe(fs.createWriteStream(dest));
+}
+
+function extractTar(source, cwd)
+{
+	const tar = require('tar');
+
+	fs.createReadStream(source).pipe(
+		tar.extract({
+			strip : 1,
+			cwd
+		})
+	);
+}
+
+function extractRemoteTar(tarUrl, cwd)
+{
+	const tar = require('tar');
+	const axios = require('axios');
```

Comment:
I'm unaware of a simple, dependency free way to make http requests. `fetch` in node is experimental in 17.5 and still hasn't entered stability in 19. 

Do you know of a way? (Simple being the key, I can use `node:http` but it would add a lot of code)

Reply:
[node-fetch](https://www.npmjs.com/package/node-fetch) is 107KB while [axios](https://www.npmjs.com/package/axios) is 1.74MB.

I thing the former is more than enough for the needed usage.

Reply:
Done! Swapped for node-fetch. Note that node-fetch >= 3 requires ESM, so I used node-fetch@2